core/domain/collection_domain.py

```diff
@@ -992,6 +1167,38 @@ def validate(self, strict=True):
         for node in self.nodes:
             node.validate()
 
+        # Validate skills.
+        for skill_id, skill in self.skills.iteritems():
+            if len(skill_id) < 6:
+                raise utils.ValidationError(
+                    'Expected skill ID to have length at least 6, received %s' %
+                    skill_id)
+
+            if skill_id[0:5] != 'skill':
+                raise utils.ValidationError(
+                    'Expected skill ID to begin with \'s\', received %s' %
+                    skill_id)
+
+            if not skill_id[5:].isdigit():
+                raise utils.ValidationError(
+                    'Expected skill ID to end with a number, received %s' %
+                    skill_id)
+
+            if int(skill_id[5:]) >= self.next_skill_id:
+                raise utils.ValidationError(
+                    'Expected skill ID number to be less than %s, received %s' %
+                    self.next_skill_id, skill_id)
+
+            skill.validate()
+
+        # Check that prerequisite and acquired skills exist in the skill table
+        for node in self.nodes:
+            for skill_id in (
+                    node.prerequisite_skill_ids + node.acquired_skill_ids):
+                if skill_id not in self.skills:
+                    raise utils.ValidationError(
+                        'Skill with ID %s does not exist' % skill_id)
+
         if strict:
```

Comment:
Do you think strict validation should include ensuring all skills are being used?